### PR TITLE
REM-650

### PR DESCRIPTION
**Pull request by RebeccaFrem02**
add a new numberfield, label = “OTP Expiry Time”

min 30, max 120, no decimal

key = otp-expiry-time

the value of this field will be used to set the timer in the sms OTP component

### CODE REVIEW

File: src/pages/ct-default/form/CtDefaults.js [hunk 1]
```diff
@@ -39,24 +28,27 @@ const CtDefaults = ({ _labels, access }) => {
...
     enableReinitialize: true,
     validateOnChange: true,
+    validationSchema: yup.object({
+      'otp-expiry-time': yup.number().min(30).max(120)
+    }),
     initialValues: {
       'ct-nra-individual': null,
-      'ct-nra-individual-ref': null,
-      'ct-nra-individual-description': null,
+      'ct-nra-individual-ref': '',
```
Comment (Ghinwa-Arab) [inline src/pages/ct-default/form/CtDefaults.js:46]: Only include the fields required by api in formik, remove:
      'ct-nra-corporate-ref': '',
      'ct-nra-corporate-description': '',
       'ct-nra-individual-ref': '',
      'ct-nra-individual-description': '',

File: src/pages/ct-default/form/CtDefaults.js [hunk 2]
```diff
@@ -70,54 +62,38 @@ const CtDefaults = ({ _labels, access }) => {
...
 
   const getDataResult = () => {
-    var parameters = `_filter=`
-    getRequest({
-      extension: CurrencyTradingSettingsRepository.Defaults.qry,
-      parameters: parameters
+    const myObject = {}
+    defaultsData.list.forEach(obj => {
+      if (arrayAllow.includes(obj.key)) {
+        myObject[obj.key] = obj.key ? parseInt(obj.value) : null
```
Comment (Ghinwa-Arab) [inline src/pages/ct-default/form/CtDefaults.js:68]: shouldn't u check obj.value insted of obj.key?

File: src/pages/ct-default/form/CtDefaults.js [hunk 3]
```diff
@@ -70,54 +62,38 @@ const CtDefaults = ({ _labels, access }) => {
...
+    if (myObject && myObject['ct-nra-corporate']) {
+      getNumberRange(myObject['ct-nra-corporate'], 'ct-nra-corporate')
+    }
   }
 
   const getNumberRange = (nraId, key) => {
     var parameters = `_filter=` + '&_recordId=' + nraId
     getRequest({
       extension: SystemRepository.NumberRange.get,
       parameters: parameters
```
Comment (Ghinwa-Arab) [inline src/pages/ct-default/form/CtDefaults.js:84]: please assign `_filter=` + '&_recordId=' + nraId' directly to request parameters no need for var parameter

File: src/pages/ct-default/form/CtDefaults.js [hunk 4]
```diff
@@ -132,54 +108,20 @@ const CtDefaults = ({ _labels, access }) => {
     await postRequest({
       extension: CurrencyTradingSettingsRepository.Defaults.set2,
       record: JSON.stringify({ sysDefaults: data })
+    }).then(res => {
+      if (res) toast.success(platformLabels.Updated)
```
Comment (Ghinwa-Arab) [inline src/pages/ct-default/form/CtDefaults.js:112]: remove if (res)

File: src/components/Shared/OTPPhoneVerification.js [hunk 5]
```diff
@@ -47,6 +49,18 @@ const OTPPhoneVerification = ({
...
 
+  useEffect(() => {
+    if (defaultsData?.list?.find(obj => obj.key === 'otp-expiry-time')) {
+      const expiryTimeObj = defaultsData.list.find(obj => obj.key === 'otp-expiry-time')
+      const expiryTime = parseInt(expiryTimeObj?.value, 10)
+      if (!isNaN(expiryTime)) {
+        setTimer(expiryTime)
+      }
+    } else {
+      setError('OTP Expiry Time is Not Filled.')
```
Comment (Ghinwa-Arab) [inline src/components/Shared/OTPPhoneVerification.js:60]: no hard code messages allowed, use translator

**Comment by NoraBitarSM** [comment]
when I click on resent OTP the timer becomes 60
but in the settings it's 40 
image

**Pull request merged**